awx/ui_next/src/screens/Organization/OrganizationList/OrganizationList.test.jsx

```diff
@@ -61,97 +63,165 @@ const mockAPIOrgsList = {
+    OrganizationsAPI.readOptions.mockResolvedValue({
+      data: {
+        actions: {
+          GET: {},
+          POST: {},
         },
-      });
-    OrganizationsAPI.readOptions = () =>
-      Promise.resolve({
-        data: {
-          actions: {
-            GET: {},
-            POST: {},
-          },
-        },
-      });
+      },
+    });
+  });
+  afterEach(() => {
+    jest.clearAllMocks();
   });
 
-  test('initially renders succesfully', () => {
-    mountWithContexts(<OrganizationsList />);
+  test('Initially renders succesfully', async () => {
+    await act(async () => {
+      mountWithContexts(<OrganizationsList />);
+    });
   });
 
-  test('Puts 1 selected Org in state when handleSelect is called.', () => {
-    wrapper = mountWithContexts(<OrganizationsList />).find(
-      'OrganizationsList'
+  test('Items are rendered after loading', async () => {
+    await act(async () => {
+      wrapper = mountWithContexts(<OrganizationsList />);
+    });
+    await waitForElement(
+      wrapper,
+      'OrganizationsList',
+      el => el.find('ContentLoading').length === 0
     );
-
-    wrapper.setState({
-      organizations: mockAPIOrgsList.data.results,
-      itemCount: 3,
-      isInitialized: true,
-    });
-    wrapper.update();
-    expect(wrapper.state('selected').length).toBe(0);
-    wrapper.instance().handleSelect(mockAPIOrgsList.data.results.slice(0, 1));
-    expect(wrapper.state('selected').length).toBe(1);
+    expect(wrapper.find('OrganizationListItem').length).toBe(3);
   });
 
-  test('Puts all Orgs in state when handleSelectAll is called.', () => {
-    wrapper = mountWithContexts(<OrganizationsList />);
-    const list = wrapper.find('OrganizationsList');
-    list.setState({
-      organizations: mockAPIOrgsList.data.results,
-      itemCount: 3,
-      isInitialized: true,
-    });
-    expect(list.state('selected').length).toBe(0);
-    list.instance().handleSelectAll(true);
-    wrapper.update();
-    expect(list.state('selected').length).toEqual(
-      list.state('organizations').length
+  test('Item appears selected after selecting it', async () => {
+    const itemCheckboxInput = 'input#select-organization-1';
+    await act(async () => {
+      wrapper = mountWithContexts(<OrganizationsList />);
+    });
+    await waitForElement(
+      wrapper,
+      'OrganizationsList',
+      el => el.find('ContentLoading').length === 0
+    );
+    await act(async () => {
+      wrapper
+        .find(itemCheckboxInput)
+        .closest('DataListCheck')
+        .props()
+        .onChange();
+    });
+    await waitForElement(
+      wrapper,
+      'OrganizationsList',
+      el => el.find(itemCheckboxInput).props().checked === true
     );
   });
 
-  test('api is called to delete Orgs for each org in selected.', () => {
-    wrapper = mountWithContexts(<OrganizationsList />);
-    const component = wrapper.find('OrganizationsList');
-    wrapper.find('OrganizationsList').setState({
```

Comment:
There's a lot of changes to this file because you can't set or check state on functional components. I've updated each test case to check the behavior (in terms of simulated input and rendering) instead of state.